firrtl/src/main/scala/firrtl/ir/IR.scala

```diff
@@ -436,6 +436,7 @@ object Formal extends Enumeration {
   pred: Expression,
   en:   Expression,
   msg:  StringLit,
+  args: Seq[Expression],
```

Comment:
If we're doing this the data class way, then this needs an `@since(...)`. @jackkoenig: how do you want to handle this?

Reply:
We've deprecated everything in package firrtl._ and we should probably make it all private as part of Chisel 7 so I'm fine with just making the breaking change. WIth that, I'm fine with this as is, or we can change it back to a case class which would let us delete the unapply (and companion object) rather than updating it.

WDYT?

Reply:
I greatly prefer making all this private!